### PR TITLE
Add serialization support for various Crypto types

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Also adds a serialization test for the Block.

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
There seem to be weird MacOS failures on GithubCI, something to do with missing symbols. And I just got a related e-mail from Github saying they are upgrading to MacOS Catalina, but we can still use Mojave. I wonder if they screwed something up. :D

**Comment by Geod24** [comment]
> And I just got a related e-mail from Github saying they are upgrading to MacOS Catalina, but we can still use Mojave. I wonder if they screwed something up. :D

No they didn't, but it means we can't use < 2.087.1 because of [this](https://dlang.org/changelog/2.087.1.html#darwin-loader).

**Comment by Geod24** [comment]
> No they didn't

Actually scratch that. It shouldn't fail on 10.14.

**Comment by Geod24** [comment]
At this point... Should we just remove the restriction ?

**Comment by AndrejMitrovic** [comment]
Restriction on what?

**Comment by Geod24** [comment]
The serialization requiring to implement `serialize` instead of just `tupleof`ing stuff.

File: source/agora/common/crypto/ECC.d [hunk 1]
```diff
@@ -84,6 +92,18 @@ public struct Scalar
...
     }
 
+    /// Ditto
+    public string toString () const @safe
+    {
+        return this.data.toString();
+    }
+
+    /// Vibe.d deserialization
+    static Scalar fromString (scope const(char)[] str) @safe
```
Comment (Geod24) [inline source/agora/common/crypto/ECC.d]:
```suggestion
    public static Scalar fromString (scope const(char)[] str) @safe
```

File: source/agora/common/crypto/ECC.d [hunk 2]
```diff
@@ -162,6 +223,18 @@ public struct Point
...
     }
 
+    /// Ditto
+    public string toString () const @safe
+    {
+        return this.data.toString();
+    }
+
+    /// Vibe.d deserialization
+    static Point fromString (scope const(char)[] str) @safe
```
Comment (Geod24) [inline source/agora/common/crypto/ECC.d]:
```suggestion
    public static Point fromString (scope const(char)[] str) @safe
```

**Review by Geod24 — APPROVED** [review]
LGTM minus nits

**Comment by AndrejMitrovic** [comment]
Fixed missing `public` attributes. Once it's green, you can merge it @TrustHenry.

**Pull request merged**